.github/workflows/tests.yml

```diff
@@ -26,11 +35,15 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip wheel setuptools tox
-      - name: Run tox targets for ${{ matrix.python-version }}
+          python -m pip install --upgrade pip wheel setuptools tox tox-factor
+      - name: Set pyversion for tox
+        run: |
+          python -c "import sys; print('::set-output name=PYV::', ''.join([str(x) for x in sys.version_info[:2]]), sep='')"
+        id: pyversion
+      - name: Run tox targets for ${{ matrix.python-version }} on ${{ matrix.os }}
         run: |
-          ENV_PREFIX=$(tr -C -d "0-9" <<< "${{ matrix.python-version }}")
-          TOXENV=$(tox --listenvs | grep "^py$ENV_PREFIX" | tr '\n' ',') python -m tox
+          python -m tox -f py${{ steps.pyversion.outputs.PYV }}-${{ matrix.os }} -l
```

Comment:
I think we can remove the `-l` step - we implicitly see it based upon what actually runs?

Reply:
Sure; I've added that primarily for debugging; took me a while to figure out what works on all three OSes and what doesn't

I'm off for today; I'll remove it tomorrow :)